### PR TITLE
Lightning threat indexes

### LINKED ISSUE #1632: Lightning Threat Indexes
## Description
The HRRR lightning threat index product must be ported to the RRFS since it is a mandatory operational product.

## Solution
Implement it.

This is already implemented; I'm in the middle of PRing it now.

## Alternatives
- Don't switch to RRFS?
- Discontinue the product?
- Pretend the problem doesn't exist?

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
## Description
Calculates the lightning threat index product from HRRR, inside CCPP and sends it to UPP for output.

The algorithm was originally ported from HRRR to the GFDL_atmos_cubed_sphere NGGPS diagnostics, using the w variable in both implementations. I've moved it to CCPP's maximum hourly diagnostics, but it still needs w in the formulation, so I've added the dynamical core's w variable as wgrs to match the u & v winds which are ugrs and vgrs.

### Input data additions/changes
- [x] No changes are expected to input data.

### Anticipated changes to regression tests:

- [x] Many regression test results may change due to updating to the latest UPP and adding fields to regional output.

## Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [x] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Combined with PR's (If Applicable):

I suggest combining this with my PR https://github.com/ufs-community/ufs-weather-model/pull/1361 which adds other UPP diagnostic output.

## Commit Queue Checklist:

- [x] Link PR's from all sub-components involved
- [x] Confirm reviews completed in sub-component PR's
- [x] Add all appropriate labels to this PR.
- [x] Run full RT suite on either Hera/Cheyenne with both Intel/GNU compilers
- [ ] Add list of any failed regression tests to "Anticipated changes to regression tests" section.

## Linked PR's and Issues:
- Depends on NOAA-EMC/fv3atm/pull/628
- Depends on ufs-community/ccpp-physics/pull/42
- Depends on NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/243
- Depends on https://github.com/NOAA-EMC/UPP/pull/673

- Closes: #1632 

## Testing Day Checklist:

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.

### Testing Log (for CM's):
- RDHPCS
  - Intel
    - [x] Hera
    - [x] Orion
    - [x] Jet
    - [ ] Gaea
    - [x] Cheyenne
  - GNU
    - [x] Hera
    - [x] Cheyenne
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment

### CODE REVIEW

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA #1547 is on hold today. So today is pretty much open day. Can we coordinate to prepare this pr? Do you want to go ahead to combine with #1361 ?

**Comment by SamuelTrahanNOAA** [comment]
The fv3atm dependency has not been approved for either PR. I shouldn't merge them until the review is complete:

https://github.com/NOAA-EMC/fv3atm/pull/628
https://github.com/NOAA-EMC/fv3atm/pull/568

**Comment by zach1221** [comment]
Hi, @SamuelTrahanNOAA . It looks like the fv3atm dependencies for this PR and 1361 have been approved. Do you feel comfortable to combine with 1361?

**Comment by SamuelTrahanNOAA** [comment]
> combine with 1361

Yes. I will merge the changes from 1361 into this PR.

**Comment by zach1221** [comment]
Thanks, @SamuelTrahanNOAA . I think we'll look to start adding baseline labels and testing of this PR next. Are you able to sync up your branch?

**Comment by SamuelTrahanNOAA** [comment]
I am testing the merged version now.

**Comment by SamuelTrahanNOAA** [comment]
> Are you able to sync up your branch?

I'm not sure what you mean. It is up-to-date with the top of the authoritative branches and the #1361 branches. Did I miss something?

**Comment by zach1221** [comment]
@SamuelTrahanNOAA No, all good. While you're testing the merged version, I'll run ORT via jenkins-ci.

**Comment by SamuelTrahanNOAA** [comment]
Someone should run the Acorn tests as soon as possible. That is the only platform with a new test, and it uses a proprietary library. I need to make sure people other than me are able to use it, and obviously that hasn't been tested with the version I just merged together a half hour ago.

**Comment by zach1221** [comment]
@BrianCurtis-NOAA are you able to begin testing on Acorn?

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA are you able to begin testing on Acorn?

Normally, yes. But, I am out today at noon, so by the time RT is created and compared with it could be 2 hours. Maybe @DusanJovic-NOAA or @DeniseWorthen or @junwang-noaa would be able to run on Acorn/WCOSS2?

File: modulefiles/ufs_acorn.intel.lua
Comment (junwang-noaa) [inline modulefiles/ufs_acorn.intel.lua]: @SamuelTrahanNOAA We can't add personal account's module path to the module file, can you put it in a group account?

Comment (SamuelTrahanNOAA) [reply modulefiles/ufs_acorn.intel.lua]: I don't have access to any group accounts.

Comment (SamuelTrahanNOAA) [reply modulefiles/ufs_acorn.intel.lua]: @WenMeng-NOAA - Do you have access to any group accounts on Acorn? If so, can you install libIFI there? I can give you instructions.

Comment (junwang-noaa) [reply modulefiles/ufs_acorn.intel.lua]: Do you know which model application will use this feature, maybe ask them to install it or give you access to install it in their group account? Is it possible that nceplibs team can install this library under nceplibs account for this PR before you get access to a group account?

Comment (SamuelTrahanNOAA) [reply modulefiles/ufs_acorn.intel.lua]: This is a proprietary library with strict controls to accessing the source code. Certainly, an nceplibs team member could do it, if they're cleared for access. However, they'll have to ensure nobody else (including other nceplibs team members) can access the source code. Wen and I have been cleared for access.

**Comment by zach1221** [comment]
jenkins-ci failed on the regional_control thr and dcp tests. I'm investigating now. The same tests passed when running manually on Hera.

File: modulefiles/ufs_acorn.intel.lua
Comment (SamuelTrahanNOAA) [reply modulefiles/ufs_acorn.intel.lua]: I don't know if he's in the the "nceplibs team" but George Vandenberghe is also cleared for access, in the original list supplied by EMC. He doesn't have access at the moment, but that is easy to fix.

**Comment by SamuelTrahanNOAA** [comment]
> jenkins-ci failed on the regional_control thr and dcp tests. I'm investigating now. The same tests passed when running manually on Hera.

There are new ifi variants of regional_control. If the naming is confusing your scripts, we can rename them.

```
COMPILE | -DAPP=ATM -DCCPP_SUITES=FV3_GFS_v16,FV3_GFS_v15_thompson_mynn,FV3_GFS_v17_p8,FV3_GFS_v17_p8_rrtmgp,FV3_GFS_v15_thompson_mynn_lam3km,FV3_WoFS_v0 -D32BIT=ON -DREQUIRE_IFI=ON | + acorn.intel | fv3 |
RUN     | regional_control_ifi                                                                                                    | + acorn.intel                           | fv3 |
RUN     | regional_decomp_ifi                                                                                                     | + acorn.intel                           |     |
RUN     | regional_2threads_ifi                                                                                                   | + acorn.intel                           |     |
```

**Comment by SamuelTrahanNOAA** [comment]
Note that the ifi tests are *only run on acorn* (`+ acorn.intel` in rt.conf) so if the jenkins ci tests on hera are running them, that's yet another bug.

**Comment by DusanJovic-NOAA** [comment]
> > @BrianCurtis-NOAA are you able to begin testing on Acorn?
> 
> Normally, yes. But, I am out today at noon, so by the time RT is created and compared with it could be 2 hours. Maybe @DusanJovic-NOAA or @DeniseWorthen or @junwang-noaa would be able to run on Acorn/WCOSS2?

I can run the tests on Acorn.

**Comment by zach1221** [comment]
Well Jenkins-ci shouldn't be running against those new ifi cases only regional_control, control_p8 and cpld_control_nowave_noaero_p8. So, when I ran ORT against this PR on Hera, I also just ran those three cases.

**Comment by SamuelTrahanNOAA** [comment]
@DusanJovic-NOAA - The NATLEV grib files should have 240 new fields at the end of the file with output from libIFI. On Acorn, about half of the levels should have a lot of non-zero non-missing values. (Everywhere else, it should be 100% missing.) If you can point me to your Acorn baseline, I can tell you if it looks right.

**Comment by DusanJovic-NOAA** [comment]
> @DusanJovic-NOAA - The NATLEV grib files should have 240 new fields at the end of the file with output from libIFI. On Acorn, about half of the levels should have a lot of non-zero non-missing values. (Everywhere else, it should be 100% missing.) If you can point me to your Acorn baseline, I can tell you if it looks right.

The new baselines, still in temporary location, are here:

/lfs/h1/emc/ptmp/dusan.jovic/FV3_RT/REGRESSION_TEST_INTEL

**Comment by SamuelTrahanNOAA** [comment]
The test data from @DusanJovic-NOAA looks good.

~~Now if someone can copy the libraries to a role account, we can proceed with the installation.~~

**Comment by SamuelTrahanNOAA** [comment]
@GeorgeVandenberghe-NOAA has moved the libraries to nceplibs on Acorn, and I've updated the paths.

File: modulefiles/ufs_acorn.intel.lua
Comment (jkbk2004) [reply modulefiles/ufs_acorn.intel.lua]: @SamuelTrahanNOAA Sounds like libIFI access issue on acorn was resolved. Can we keep proceeding to work on this pr? @zach1221 FYI

Comment (WenMeng-NOAA) [reply modulefiles/ufs_acorn.intel.lua]: I don't have group account on Acorn. @SamuelTrahanNOAA

Comment (SamuelTrahanNOAA) [reply modulefiles/ufs_acorn.intel.lua]: It's okay. George copied it to the nceplibs area, and I updated the modulefiles. I think we're meeting the library requirements now.

Comment (jkbk2004) [reply modulefiles/ufs_acorn.intel.lua]: @SamuelTrahanNOAA another question. so people with access to acorn group account can run the ifi cases on acorn, right?

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA We can start working on this pr. Please, go ahead to sync up.

**Comment by SamuelTrahanNOAA** [comment]
I have merged the latest changes. It appears the only changes were in AQM, so I don't expect this to affect my developments.

**Comment by zach1221** [comment]
Jenkins-ci still failing on the regional_control dcp and thr Operational Requirement Tests. Running these against this PR on Hera again now to ensure it at least passes there still.

**Comment by jkbk2004** [comment]
@zach1221 can we confirm again regional_control fails on jenkins-ci?

**Comment by SamuelTrahanNOAA** [comment]
I should try renaming regional_control_ifi to regional_ifi_control to see if that fixes the tests. It may be picking up `_ifi` as a variant instead of an entirely different test.

**Comment by SamuelTrahanNOAA** [comment]
I have renamed the `regional_*_ifi` tests to `regional_ifi_*`. Could you please try again and see if it fixed the problems?

**Comment by zach1221** [comment]
Thanks, @SamuelTrahanNOAA . Trying again now.

**Comment by zach1221** [comment]
Regional_control still fails in the jenkins pipeline, however the tests pass on Hera. @jkbk2004 does cmp_proc_bind.sh need to be updated maybe?

**Comment by jkbk2004** [comment]
jenkins gnu is catching something again, I think.

**Comment by zach1221** [comment]
Yes, hera gnu is failing also when running Regional_control ORT. 
[err.txt](https://github.com/ufs-community/ufs-weather-model/files/11034336/err.txt)

**Comment by SamuelTrahanNOAA** [comment]
I think you're hitting a stack limit. That line declares some large local arrays:

```f90
      SUBROUTINE CALSLR_ROEBBER(tprs,rhprs,slr)
...
      type all_grids
           real :: grid
           real :: sigma
      end type all_grids
...
      type(all_grids), dimension(ista:iend,jsta:jend,0:lsm) :: tmpk_grids, rh_grids
```

**Comment by jkbk2004** [comment]
> I think you're hitting a stack limit. That line declares some large local arrays:
> 
> ```fortran-modern
>       SUBROUTINE CALSLR_ROEBBER(tprs,rhprs,slr)
> ...
>       type all_grids
>            real :: grid
>            real :: sigma
>       end type all_grids
> ...
>       type(all_grids), dimension(ista:iend,jsta:jend,0:lsm) :: tmpk_grids, rh_grids
> ```

@WenMeng-NOAA By any chance, any idea about the error above?

**Comment by WenMeng-NOAA** [comment]
> > I think you're hitting a stack limit. That line declares some large local arrays:
> > ```
> >       SUBROUTINE CALSLR_ROEBBER(tprs,rhprs,slr)
> > ...
> >       type all_grids
> >            real :: grid
> >            real :: sigma
> >       end type all_grids
> > ...
> >       type(all_grids), dimension(ista:iend,jsta:jend,0:lsm) :: tmpk_grids, rh_grids
> > ```
> 
> @WenMeng-NOAA By any chance, any idea about the error above?

It seems the inline post crashed again at calculation of snow liquid ratio (SLR). I will reach out to the developer to look into it.

**Comment by SamuelTrahanNOAA** [comment]
I found an error in CALSLR_ROEBBER in UPP_PHYSICS.f that would cause this. It is sending an array with the wrong bounds to the BOUND function, causing it to step all over memory.

```f90
real,dimension(ista:iend,jsta:jend)    :: T2M,RH2M
...
CALL BOUND(RH2M,H1,H100)
```

The BOUND subroutine expects an IM,JM array:

```f90
SUBROUTINE BOUND(FLD,FMIN,FMAX)
REAL,intent(in) :: FMAX, FMIN
REAL,intent(inout) :: FLD(IM,JM)
```

**Comment by SamuelTrahanNOAA** [comment]
I opened an issue here:

https://github.com/NOAA-EMC/UPP/issues/672

I'm testing a fix now.

**Comment by SamuelTrahanNOAA** [comment]
Fix is here: https://github.com/NOAA-EMC/UPP/pull/673

**Comment by SamuelTrahanNOAA** [comment]
@jkbk2004 @zach1221 - Please test this branch again and see if my UPP fix lets the tests pass.

**Comment by WenMeng-NOAA** [comment]
> Fix is here: [NOAA-EMC/UPP#673](https://github.com/NOAA-EMC/UPP/pull/673)

@SamuelTrahanNOAA Thanks for fixing UPP bug. I will proceed the UPP update.

**Comment by SamuelTrahanNOAA** [comment]
I just updated my UPP branch to the top of UPP's develop, which may affect something. You will need to test again. It may be wise to wait until the gnu test succeeds or fails, so we can answer the crashing question sooner.

**Comment by zach1221** [comment]
@SamuelTrahanNOAA @jkbk2004 Hera.gnu failed with a different error.
[err2.txt](https://github.com/ufs-community/ufs-weather-model/files/11040375/err2.txt)

**Comment by SamuelTrahanNOAA** [comment]
What you saw is not an error and will not cause the job to hang. It is a warning that most of the jobs should print (intentional part of the test). Something else caused the job to freeze until it hit its wallclock limit. Do you have any other diagnostics? Perhaps the "out" file?

> This post cannot produce IFI icing products because it was not compiled with libIFI

(Not an error; should be printed.)

**Comment by SamuelTrahanNOAA** [comment]
Running a different test system than everyone else makes it harder to debug. You should update the rt.sh to run what you run, or it'll be impossible to make any progress.

**Comment by WenMeng-NOAA** [comment]
> I found an error in CALSLR_ROEBBER in UPP_PHYSICS.f that would cause this. It is sending an array with the wrong bounds to the BOUND function, causing it to step all over memory.
> 
> ```fortran-modern
> real,dimension(ista:iend,jsta:jend)    :: T2M,RH2M
> ...
> CALL BOUND(RH2M,H1,H100)
> ```
> 
> The BOUND subroutine expects an IM,JM array:
> 
> ```fortran-modern
> SUBROUTINE BOUND(FLD,FMIN,FMAX)
> REAL,intent(in) :: FMAX, FMIN
> REAL,intent(inout) :: FLD(IM,JM)
> ```

@SamuelTrahanNOAA The routine BOUND is called across UPP. Do you think redefining FLD in BOUND  might reduce similar memory exhausted issues in UPP?
```
change
  REAL,intent(inout) :: FLD(IM,JM)
into
   REAL,intent(inout) :: FLD(ista:iend,jst:jend)
```

**Comment by SamuelTrahanNOAA** [comment]
I can't even see the log files from your runs. This new test system will slow down the commit process, not speed it up.

**Comment by SamuelTrahanNOAA** [comment]
> The routine BOUND is called across UPP. Do you think redefining FLD in BOUND might reduce similar memory exhausted issues in UPP?

No, then it'll be wrong in all the places that were calling it correctly before.

**Comment by SamuelTrahanNOAA** [comment]
I found another RH out-of-bounds access. I don't think this would cause the job to freeze until its wallclock limit though; something else is wrong.

I'd also like to say that this is the *fourth* UPP bug I have had to fix for this PR, that is unrelated to my changes.

**Comment by SamuelTrahanNOAA** [comment]
I made an issue for this bug: https://github.com/NOAA-EMC/UPP/issues/674 and I'm going to put the fix in the same UPP branch.

**Comment by SamuelTrahanNOAA** [comment]
The fix is in the branch, but I want to see the `out` from the failed job before you rerun, so I can hopefully determine why it failed.

**Comment by zach1221** [comment]
@SamuelTrahanNOAA the out.log for my ORT job on Hera doesn't capture a UPP error, it just hangs and fails.
[1642_ORTout.txt](https://github.com/ufs-community/ufs-weather-model/files/11040699/1642_ORTout.txt)

Perhaps when the jenkins job finishes, it was show something other than complaining about ifi.

**Comment by jkbk2004** [comment]
@zach1221 can you test again with jenkins ci sandbox: regional_control ?

**Comment by SamuelTrahanNOAA** [comment]
> Perhaps when the jenkins job finishes, it was show something other than complaining about ifi.

It looks to me like it ran slower than usual and hit its wallclock limit. I don't see any indications that anything went wrong.

**Comment by jkbk2004** [comment]
> > Perhaps when the jenkins job finishes, it was show something other than complaining about ifi.
> 
> It looks to me like it ran slower than usual and hit its wallclock limit. I don't see any indications that anything went wrong.

I agree hera.gnu ORT for regional_control is not standard (it can be slow). We can confirm when @zach1221 brings jenkin-ci test for the case.

**Comment by zach1221** [comment]
The latest jenkins-ci run has finally finished and regional_control ORT passed Ok this time. 
[ufs-weather-model » ort-docker-pipeline » PR-1642 #4 Console [Jenkins].pdf](https://github.com/ufs-community/ufs-weather-model/files/11041496/ufs-weather-model.ort-docker-pipeline.PR-1642.4.Console.Jenkins.pdf)

**Comment by SamuelTrahanNOAA** [comment]
Please wait until @WenMeng-NOAA finishes her code review before running the final tests.

**Comment by WenMeng-NOAA** [comment]
@SamuelTrahanNOAA Your UPP PR was merged in the UPP develop at 22cfb88.

**Comment by SamuelTrahanNOAA** [comment]
My UPP bug fixes are in develop now, and my hera gnu tests passed again.

**Comment by jkbk2004** [comment]
All sounds good! I will start adding test labels after lunch.

**Comment by SamuelTrahanNOAA** [comment]
I think I was unclear. The upp is merged, but the other three PRs aren't, yet. They've been approved, but nobody pressed the "merge" buttons yet.

https://github.com/NOAA-EMC/fv3atm/pull/628
https://github.com/ufs-community/ccpp-physics/pull/42
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/243

**Comment by jkbk2004** [comment]
> I think I was unclear. The upp is merged, but the other three PRs aren't, yet. They've been approved, but nobody pressed the "merge" buttons yet.
> 
> [NOAA-EMC/fv3atm#628](https://github.com/NOAA-EMC/fv3atm/pull/628) [ufs-community/ccpp-physics#42](https://github.com/ufs-community/ccpp-physics/pull/42) [NOAA-GFDL/GFDL_atmos_cubed_sphere#243](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/243)

Not necessary to require merging dependencies before full tests. But upp side followed its own test results and approvals. It's safe we ask merging other dependencies after we fully finish tests.

**Comment by SamuelTrahanNOAA** [comment]
> Not necessary to require merging dependencies before full tests. But upp side followed its own test results and approvals. It's safe we ask merging other dependencies after we fully finish tests.

Sounds good to me. I'll keep my fingers crossed and hope for no more UPP bugs.

**Comment by support-epic** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
Repo location: /glade/scratch/epicufsrt/autort/jenkins/autort/pr/1265303346/20230322113710/ufs-weather-model
Please manually delete: /glade/scratch/epicufsrt/FV3_RT/rt_25087
Test compile_030 failed in run_compile failed 
Test compile_033 failed in run_compile failed 
Please make changes and add the following label back: cheyenne-intel-BL

**Comment by SamuelTrahanNOAA** [comment]
The post from @support-epic is incorrect. I just looked at the log files from the two "failed" jobs:

```sh
/glade/scratch/epicufsrt/FV3_RT/rt_25087/compile_{030,033}/{out,err}
```

and both compilations succeeded.

```sh
make[2]: Leaving directory '/glade/scratch/epicufsrt/FV3_RT/rt_25087/compile_030/build_fv3_030'
[100%] Built target ufs_model
make[1]: Leaving directory '/glade/scratch/epicufsrt/FV3_RT/rt_25087/compile_030/build_fv3_030'
/glade/u/apps/ch/opt/cmake/3.22.0/bin/cmake -E cmake_progress_start /glade/scratch/epicufsrt/FV3_RT/rt_25087/compile_030/build_fv3_030/CMakeFiles 0
Elapsed time 894 seconds. Compiling -DAPP=ATMAERO -DCCPP_SUITES=FV3_GFS_v17_p8 -D32BIT=ON -DMPI=ON -DCMAKE_BUILD_TYPE=Release finished
Compile ended:     Wed Mar 22 12:26:25 MDT 2023
```

```sh
+ rm -rf /glade/scratch/epicufsrt/FV3_RT/rt_25087/compile_030/build_fv3_030
+ elapsed=894
+ echo 'Elapsed time 894 seconds. Compiling -DAPP=ATMAERO -DCCPP_SUITES=FV3_GFS_v17_p8 -D32BIT=ON -DMPI=ON -DCMAKE_BUILD_TYPE=Release finished'
+ echo 'Compile 030 elapsed time 894 seconds. -DAPP=ATMAERO -DCCPP_SUITES=FV3_GFS_v17_p8 -D32BIT=ON -DMPI=ON -DCMAKE_BUILD_TYPE=Release'
++ date
+ echo 'Compile ended:    ' Wed Mar 22 12:26:25 MDT 2023
++ date +%s
+ echo -n ' 1679509585,'
```

```sh
[100%] Built target ufs_model
make[1]: Leaving directory '/glade/scratch/epicufsrt/FV3_RT/rt_25087/compile_033/build_fv3_033'
/glade/u/apps/ch/opt/cmake/3.22.0/bin/cmake -E cmake_progress_start /glade/scratch/epicufsrt/FV3_RT/rt_25087/compile_033/build_fv3_033/CMakeFiles 0
Elapsed time 1016 seconds. Compiling -DAPP=ATMAQ -DCCPP_SUITES=FV3_GFS_v15p2 -DFASTER=ON -D32BIT=ON -DMPI=ON -DCMAKE_BUILD_TYPE=Release finished
Compile ended:     Wed Mar 22 12:34:21 MDT 2023
```

```sh
+ rm -rf /glade/scratch/epicufsrt/FV3_RT/rt_25087/compile_033/build_fv3_033
+ elapsed=1016
+ echo 'Elapsed time 1016 seconds. Compiling -DAPP=ATMAQ -DCCPP_SUITES=FV3_GFS_v15p2 -DFASTER=ON -D32BIT=ON -DMPI=ON -DCMAKE_BUILD_TYPE=Release finished'
+ echo 'Compile 033 elapsed time 1016 seconds. -DAPP=ATMAQ -DCCPP_SUITES=FV3_GFS_v15p2 -DFASTER=ON -D32BIT=ON -DMPI=ON -DCMAKE_BUILD_TYPE=Release'
++ date
+ echo 'Compile ended:    ' Wed Mar 22 12:34:21 MDT 2023
++ date +%s
+ echo -n ' 1679510061,'
```

Perhaps your error detection system is erroneous?

**Comment by zach1221** [comment]
@SamuelTrahanNOAA sometimes there a timeout related failures. I'll investigate and try running those tests manually.

**Comment by BrianCurtis-NOAA** [comment]
Were the GNU/INTEL RT's run to show which tests fail?

**Comment by SamuelTrahanNOAA** [comment]
> Were the GNU/INTEL RT's run to show which tests fail?

Not with any recent version of the branch, and there have been many UPP bug fixes since then. Do you want me to do that now?

**Comment by BrianCurtis-NOAA** [comment]
> > Were the GNU/INTEL RT's run to show which tests fail?
> 
> Not with any recent version of the branch, and there have been many UPP bug fixes since then. Do you want me to do that now?

No, since you mention it's going to be a lot, i think Jong just created the whole RT suite worth of new baselines anyways.

**Comment by SamuelTrahanNOAA** [comment]
We can't afford any delay in the commit queue, despite GAEA being down. I suggest we merge #1636 into this PR and restart testing. That PR will also change many test results, so there isn't much loss of testing quality by merging it into this one.

**Comment by jkbk2004** [comment]
> We can't afford any delay in the commit queue, despite GAEA being down. I suggest we merge #1636 into this PR and restart testing. That PR will also change many test results, so there isn't much loss of testing quality by merging it into this one.

@SamuelTrahanNOAA I agree to skip gaea. Once wcoss2 test are done, we can merge this pr and move on to #1636.

**Comment by climbfuji** [comment]
FYI gaea is up again since this morning

**Review by DusanJovic-NOAA — DISMISSED** [review]

**Comment by jkbk2004** [comment]
OK! all tests are done. We can start merging process.

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA I will go to ask to merge upstream dependencies.

**Comment by jkbk2004** [comment]
BTW, Gaea C4 jobs are not moving at all. An email sent out to Gaea admin.

**Comment by jkbk2004** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
[BL] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/1265303346/20230323031507/ufs-weather-model
Please make changes and add the following label back: gaea-intel-BL

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA all set! can you update fv3 pointer and revert changes in gitmodules?

**Comment by SamuelTrahanNOAA** [comment]
The fv3atm should be right now.

**Review by jkbk2004 — APPROVED** [review]

**Comment by SamuelTrahanNOAA** [comment]
I had forgotten to push that change. Now that the acorn modulefiles are updated, I suggest you rerun the regression tests there, to make sure it runs in your account and reproduces the baseline.

**Comment by BrianCurtis-NOAA** [comment]
I'm fairly certain I ran with the updated modulefiles. I tried a pull and it did not pick up any modulefiles changes and the change shows in my clone. No need to re-run on Acorn.

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA I agree. PR is ready to merge.

**Comment by jkbk2004** [comment]
Please, go ahead to approve the PR.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by grantfirl** [comment]
@SamuelTrahanNOAA Do you have any guidance to help us understand which RTs are expected to fail with this? We're moving these changes to the NCAR fork (https://github.com/NCAR/ufs-weather-model/pull/92) and re-running RTs there and we'd like to understand which tests should fail with the UPP and regional output changes.

**Comment by SamuelTrahanNOAA** [comment]
> @SamuelTrahanNOAA Do you have any guidance to help us understand which RTs are expected to fail with this? We're moving these changes to the NCAR fork (https://github.com/NCAR/ufs-weather-model/pull/92) and re-running RTs there and we'd like to understand which tests should fail with the UPP and regional output changes.

When one updates UPP, any GRIB files may change output. The `regional*` tests will also change due to the new diag_table. Put together, that's nearly all of the tests.

**Pull request merged**
